### PR TITLE
execution name fix to avoid dropping message with bad cumulus_meta

### PR DESCRIPTION
**Pull request by nemreid**
**Summary:** Summary of changes

Addresses [CUMULUS-XX: Develop amazing new feature](https://bugs.earthdata.nasa.gov/browse/CUMULUS-XXX)

## Changes

* Detailed list or prose of changes
* ...

## PR Checklist

- [ ] Update CHANGELOG
- [ ] Unit tests
- [ ] Adhoc testing
- [ ] Integration tests

### CODE REVIEW

**Review by botanical — APPROVED** [review]
Looks good to me

**Comment by nemreid** [comment]
@botanical I've updated it based on some findings of actual messages I saw being written to my dead letter archive, to add unwrapping logic. Please re-review when able.

File: packages/api/tests/lambdas/test-write-db-dlq-records-to-s3.js [hunk 1]
```diff
@@ -103,3 +107,89 @@ test.serial('write-db-dlq-records-to-s3 throws error if system bucket is not def
...
+    },
+  };
+  const testSqsMessage = {
+    sourceEvent: 'aws:sqs',
+    body: JSON.stringify(testStatesMessage),
+  };
+  t.deepEqual(unwrapDeadLetterCumulusMessage(testSqsMessage), cumulusMessage);
+});
+
+test('unwrapDeadLetterCumulusMessage returns an non-unwrappable message', (t) => {
```
Comment (botanical) [inline packages/api/tests/lambdas/test-write-db-dlq-records-to-s3.js:187]: 🐜 
```suggestion
test('unwrapDeadLetterCumulusMessage returns input as-is if it cannot be unwrapped', (t) => {
```
or something like that? 🤔

**Review by botanical — COMMENTED** [review]
I'm not sure if this case is possible but what if we have:
```
const testStatesMessage = {
    source: 'aws.states',
    detail: {
      output: null,
    },
  };
  ```
  
  Then `unwrappedDeadLetterCumulusMessage` will return:
  `JSON.parse(output || input);
Uncaught SyntaxError: Unexpected token u in JSON at position 0`
🤔 

So it looks like these are the different execution events that can occur:
- started
- succeeded
- failed
- timed out
- aborted

and no input looks like it's just an empty object string `"{}"`

so perhaps the test case above is not possible.

**Review by botanical — APPROVED** [review]
Approved but I had one concern in the comment I left

**Pull request merged**